lib/compat/wordpress-6.2/get-global-styles-and-settings.php

```diff
@@ -189,3 +189,38 @@ function _gutenberg_get_global_stylesheet_clean_cache_upon_upgrading( $upgrader,
 	}
 }
 
+/*
+ * Function to get the settings resulting of merging core, theme, and user data.
+ *
+ * @param array $path    Path to the specific setting to retrieve. Optional.
+ *                       If empty, will return all settings.
+ * @param array $context {
+ *     Metadata to know where to retrieve the $path from. Optional.
+ *
+ *     @type string $block_name Which block to retrieve the settings from.
+ *                              If empty, it'll return the settings for the global context.
+ *     @type string $origin     Which origin to take data from.
+ *                              Valid values are 'all' (core, theme, and user) or 'base' (core and theme).
+ *                              If empty or unknown, 'all' is used.
+ * }
+ *
+ * @return array The settings to retrieve.
+ */
+function gutenberg_get_global_settings( $path = array(), $context = array() ) {
+	if ( ! empty( $context['block_name'] ) ) {
+		$path = array_merge( array( 'blocks', $context['block_name'] ), $path );
+	}
+
+	$origin = 'custom';
+	if ( ! wp_theme_has_theme_json() ) {
+		// For themes with no theme.json skip querying the database for user data (custom origin).
+		$origin = 'theme';
+	} elseif ( isset( $context['origin'] ) && 'base' === $context['origin'] ) {
```

Comment:
Maybe combine this into a single `if` clause with `||`, since the logic within is the same.

Reply:
Addressed https://github.com/WordPress/gutenberg/pull/45971/commits/1a5bbe8bd127051d0ea44ad42459e8c18101a073